test/jetstream/api/object_test.exs

```diff
@@ -0,0 +1,218 @@
+      assert {:ok, _object} = Object.put_object(:gnat, bucket, "SOMETHING.md", io)
+
+      assert {:ok, objects} = Object.list_objects(:gnat, bucket)
+      [readme, something] = Enum.sort_by(objects, & &1.name)
+      assert readme.name == "README.md"
+      assert readme.size == something.size
+      assert readme.digest == something.digest
+
+      assert :ok = Object.delete_bucket(:gnat, bucket)
+    end
+  end
+
+  describe "put_object/4" do
+    test "creates an object" do
+      assert {:ok, %{config: _stream}} = Object.create_bucket(:gnat, "MY-STORE")
+
+      expected_sha = @readme_path |> File.read!() |> then(&:crypto.hash(:sha256, &1))
+      assert {:ok, object_meta} = put_filepath(@readme_path, "MY-STORE", "README.md")
+      assert object_meta.name == "README.md"
+      assert object_meta.bucket == "MY-STORE"
+      assert object_meta.chunks == 1
+      assert "SHA-256=" <> encoded = object_meta.digest
+      assert Base.url_decode64!(encoded) == expected_sha
+
+      assert :ok = Object.delete_bucket(:gnat, "MY-STORE")
+    end
+
+    test "overwriting a file" do
+      bucket = nuid()
+      assert {:ok, %{config: _stream}} = Object.create_bucket(:gnat, bucket)
+      assert {:ok, _} = put_filepath(@readme_path, bucket, "WAT")
+      size_after_readme = stream_byte_size(bucket)
+      assert {:ok, _} = put_filepath(@changelog_path, bucket, "WAT")
+      size_after_changelog = stream_byte_size(bucket)
+      assert size_after_changelog < size_after_readme
+      assert {:ok, [meta]} = Object.list_objects(:gnat, bucket)
+      assert meta.name == "WAT"
+    end
+
+    test "return an error if the object store doesn't exist" do
+      assert {:error, err} = put_filepath(@readme_path, "I_DONT_EXIST", "foo")
+      assert %{"code" => 404, "description" => "stream not found"} = err
+    end
+  end
+
+  test "storing and retrieving larger files" do
+    assert {:ok, path, sha} = generate_big_file()
+    bucket = nuid()
+    assert {:ok, %{config: _stream}} = Object.create_bucket(:gnat, bucket)
+    assert {:ok, meta} = put_filepath(path, bucket, "big")
+    assert meta.chunks == 16
+    assert meta.size == 16 * 128 * 1024
+    assert "SHA-256=" <> encoded = meta.digest
+    assert Base.url_decode64!(encoded) == sha
+
+    Process.put(:buffer, "")
+
+    Object.get_object(:gnat, bucket, "big", fn chunk ->
+      Process.put(:buffer, Process.get(:buffer) <> chunk)
+    end)
+
+    file_contents = Process.get(:buffer)
+    assert byte_size(file_contents) == meta.size
+    assert :crypto.hash(:sha256, file_contents) == sha
+    assert stream_byte_size(bucket) > 2 * 1024 * 1024
+
+    assert :ok = Object.delete_object(:gnat, bucket, "big")
+    assert stream_byte_size(bucket) < 1024
+    :ok = Object.delete_bucket(:gnat, bucket)
+  end
+
+  # create a random 2MB binary file
+  # re-use it on subsequent test runs if it already exists
+  defp generate_big_file do
+    filepath = Path.join("tmp", "big_file.bin")
```

Comment:
Could we us the ExUnit [tmp_dir](https://hexdocs.pm/ex_unit/main/ExUnit.Case.html#module-tmp-dir) here?